src/scan/postgres_scan.cpp

```diff
@@ -13,6 +15,29 @@ namespace pgduckdb {
 // PostgresScanGlobalState
 //
 
+void
+PostgresScanGlobalState::ConstructQueryFilter(duckdb::TableFilter *filter, const char *column_name) {
+	switch (filter->filter_type) {
+	case duckdb::TableFilterType::CONSTANT_COMPARISON:
+	case duckdb::TableFilterType::IS_NULL:
+	case duckdb::TableFilterType::IS_NOT_NULL:
+	case duckdb::TableFilterType::CONJUNCTION_OR:
+	case duckdb::TableFilterType::CONJUNCTION_AND:
+	case duckdb::TableFilterType::IN_FILTER:
+		scan_query << filter->ToString(column_name).c_str();
+		break;
+	case duckdb::TableFilterType::OPTIONAL_FILTER: {
+		auto optional_filter = reinterpret_cast<duckdb::OptionalFilter *>(filter);
+		ConstructQueryFilter(optional_filter->child_filter.get(), column_name);
+		break;
+	}
+	case duckdb::TableFilterType::STRUCT_EXTRACT:
+	case duckdb::TableFilterType::DYNAMIC_FILTER:
+		scan_query << "1 = 1";
```

Comment:
This seems like a bit of a hack. Do we really need this? If so, can you add a column explaining why?

Also, is this actually correct? Because now we don't actually filter anything for these filter types inside postgres, but duckdb will not filter on that side either. I feel like we probably need some tests that show that we return correct result when using these kind of things.

Reply:
Updated construction of pushdown filters. Looking at duckdb code `STRUCT_EXTRACT` is only created if `struct_extract` is used in query, while `DYNAMIC_FILTER` is constructed when there is topN node (LIMIT). Both of these should not matter to filtering.